### PR TITLE
Project cleanup

### PR DESCRIPTION
**Pull request by sgdc3**
- Maven cleanup, define plugin versions explicitly (maven uses older versions by default), inherit plugins from parent, add project info to the parent pom
- Use repo.codemc.org as maven repository
- Update bukkit and bungee dependencies to 1.13.2
- Drop java 1.7 support (less than 0.7% of servers), use labdas where possible
- Update .gitignore for more IDEs and evnironments
- General codestyle uniformation

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,13 +1,86 @@
-# compile output
+# Git
```
Comment (DoNotSpamPls) [inline .gitignore]: I'm against adding too much things in .gitignore, after all the project is not that big. If someone's configuration generates weird files, they can just add those if needed.

File: bstats-bukkit-lite/pom.xml [hunk 2]
```diff
@@ -2,63 +2,16 @@
...
         <artifactId>bstats</artifactId>
         <groupId>org.bstats</groupId>
-        <version>1.3</version>
+        <version>1.4-SNAPSHOT</version>
     </parent>
-    <modelVersion>4.0.0</modelVersion>
 
     <artifactId>bstats-bukkit-lite</artifactId>
 
-    <properties>
```
Comment (DoNotSpamPls) [inline bstats-bukkit-lite/pom.xml]: Why were these removed??

File: bstats-bukkit-lite/src/main/java/org/bstats/bukkit/MetricsLite.java [hunk 3]
```diff
@@ -111,7 +107,8 @@ public MetricsLite(Plugin plugin) {
             ).copyDefaults(true);
             try {
                 config.save(configFile);
-            } catch (IOException ignored) { }
+            } catch (IOException ignored) {
```
Comment (DoNotSpamPls) [inline bstats-bukkit-lite/src/main/java/org/bstats/bukkit/MetricsLite.java]: While there's not anything particularly wrong with this layout, I don't think it's something that should be changed.

File: bstats-bukkit-lite/src/main/java/org/bstats/bukkit/MetricsLite.java [hunk 4]
```diff
@@ -273,7 +281,7 @@ public void run() {
      * Sends the data to the bStats server.
      *
      * @param plugin Any plugin. It's just used to get a logger instance.
-     * @param data The data to send.
+     * @param data   The data to send.
```
Comment (DoNotSpamPls) [inline bstats-bukkit-lite/src/main/java/org/bstats/bukkit/MetricsLite.java]: weird space

File: pom.xml [hunk 5]
```diff
@@ -6,13 +6,13 @@
...
     <artifactId>bstats</artifactId>
+    <version>1.4-SNAPSHOT</version>
     <packaging>pom</packaging>
-    <version>1.3</version>
 
-    <properties>
-        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
-    </properties>
+    <name>BStats-Metrics</name>
```
Comment (DoNotSpamPls) [inline pom.xml]: The project's name is written `bStats`, not `BStats`

File: pom.xml [hunk 6]
```diff
@@ -23,47 +23,166 @@
         <module>bstats-sponge-lite</module>
     </modules>
 
+    <scm>
+        <connection>scm:git:https://github.com/Bastian/bStats-Metrics.git</connection>
+        <developerConnection>scm:git:git@github.com:Bastian/bStats-Metrics.git</developerConnection>
+        <url>https://github.com/Bastian/bStats-Metrics</url>
+    </scm>
+
+    <ciManagement>
```
Comment (DoNotSpamPls) [inline pom.xml]: This is not even officially used by Bastian, why are you adding it? Same for the distribution management down below - bStats-Metrics has already moved to jitpack.io

File: bstats-bukkit-lite/src/main/java/org/bstats/bukkit/MetricsLite.java [hunk 7]
```diff
@@ -158,7 +165,7 @@ public void run() {
                     }
```
Comment (DoNotSpamPls) [inline bstats-bukkit-lite/src/main/java/org/bstats/bukkit/MetricsLite.java]: If Java 7 support is removed, might as well use lambdas here too.

File: bstats-bukkit-lite/pom.xml [hunk 8]
```diff
@@ -70,9 +23,9 @@
...
             <groupId>org.bukkit</groupId>
             <artifactId>bukkit</artifactId>
-            <version>1.12.2-R0.1-SNAPSHOT</version>
+            <version>1.13.2-R0.1-SNAPSHOT</version>
             <scope>provided</scope>
         </dependency>
     </dependencies>
 
-</project>
+</project>
```
Comment (mcmonkey4eva) [inline bstats-bukkit-lite/pom.xml]: ^ You've discarded the newline-at-end-of-file from quite a few files in this PR. As you can see by the little red symbol here, GitHub renders this explicitly as a mistake (per general code practice, all code files should end with a newline unless there's a very specific reason to not have that).

Comment (DoNotSpamPls) [reply bstats-bukkit-lite/pom.xml]: Interestingly enough, my GitHub doesn't want to show that symbol. Weird

Comment (mcmonkey4eva) [reply bstats-bukkit-lite/pom.xml]: It only displays on some versions of the page (I don't know what logic GitHub uses for it), mainly the full-commit view (or full-PR-change view) show it. Just for reference, this is how GitHub shows it:
No-newline-at-EOL

File: bstats-bukkit-lite/src/main/java/org/bstats/bukkit/MetricsLite.java (hunk 3 above)
Comment (sgdc3) [reply bstats-bukkit-lite/src/main/java/org/bstats/bukkit/MetricsLite.java]: yeah, i missed that, i did an automatic codestyle cleanup based on the general codestyle

File: bstats-bukkit-lite/src/main/java/org/bstats/bukkit/MetricsLite.java (hunk 4 above)
Comment (sgdc3) [reply bstats-bukkit-lite/src/main/java/org/bstats/bukkit/MetricsLite.java]: intellij is love

File: pom.xml (hunk 5 above)
Comment (sgdc3) [reply pom.xml]: right

File: pom.xml (hunk 6 above)
Comment (sgdc3) [reply pom.xml]: As i talked privately with bastian, jitpack is causing some issues in some scenarios.

File: bstats-bukkit-lite/src/main/java/org/bstats/bukkit/MetricsLite.java (hunk 7 above)
Comment (sgdc3) [reply bstats-bukkit-lite/src/main/java/org/bstats/bukkit/MetricsLite.java]: hmm, missed that

File: bstats-bukkit-lite/pom.xml (hunk 8 above)
Comment (sgdc3) [reply bstats-bukkit-lite/pom.xml]: The majority of the files in the projects doesn't use newline at the end of the file, i just uniformed the codestyle.

File: .gitignore (hunk 1 above)
Comment (sgdc3) [reply .gitignore]: roger

File: bstats-bukkit-lite/pom.xml (hunk 8 above)
Comment (sgdc3) [reply bstats-bukkit-lite/pom.xml]: Yeah, i'm a Newline-at-EOF guy, but i was concerned bastian had a different opinion on that

File: bstats-sponge-lite/src/main/java/org/bstats/sponge/MetricsLite2.java [hunk 9]
```diff
@@ -287,13 +277,13 @@ private void startSubmitting() {
                                     }
                                 }
                             }
-                        } catch (Exception e) {
+                        } catch (Exception ignored) {
```
Comment (DoNotSpamPls) [inline bstats-sponge-lite/src/main/java/org/bstats/sponge/MetricsLite2.java]: Here `{ }` should be used as well for consistency

File: pom.xml [hunk 10]
```diff
@@ -6,13 +6,13 @@
...
+    <version>1.4-SNAPSHOT</version>
     <packaging>pom</packaging>
-    <version>1.3</version>
 
-    <properties>
-        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
-    </properties>
+    <name>bStats-Metrics</name>
+    <description>The BStats metrics classes for various server implementations</description>
```
Comment (DoNotSpamPls) [inline pom.xml]: Again, bStats instead of BStats

File: pom.xml (hunk 6 above)
Comment (DoNotSpamPls) [reply pom.xml]: Could you describe those issues in more detail? I haven't had any for the moment.

**Comment by sgdc3** [comment]
@DoNotSpamPls done

**Comment by Bastian** [comment]
Lgtm, thx 👍
